sdk/storage/azure-storage-queue/tests/test_queue_async.py

```diff
@@ -145,6 +148,156 @@ async def test_create_queue_with_options(self, storage_account_name, storage_acc
+        await queue_client.send_message(u'message9')
+        await queue_client.send_message(u'message10')
+        result = []
+        async for m in queue_client.receive_messages(max_messages=5):
+            result.append(m)
+
+        # Asserts
+        self.assertIsNotNone(result)
+        self.assertEqual(5, len(result))
+
+        for message in result:
+            self.assertIsNotNone(message)
+            self.assertNotEqual('', message.id)
+            self.assertNotEqual('', message.content)
+            self.assertNotEqual('', message.pop_receipt)
+            self.assertEqual(1, message.dequeue_count)
+            self.assertNotEqual('', message.inserted_on)
+            self.assertNotEqual('', message.expires_on)
+            self.assertNotEqual('', message.next_visible_on)
+
+    @QueuePreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_get_messages_with_too_little_messages(self, storage_account_name, storage_account_key):
+        # Action
+        qsc = QueueServiceClient(self.account_url(storage_account_name, "queue"), storage_account_key)
+        queue_client = self._get_queue_reference(qsc)
+        await queue_client.create_queue()
+        await queue_client.send_message(u'message1')
+        await queue_client.send_message(u'message2')
+        await queue_client.send_message(u'message3')
+        await queue_client.send_message(u'message4')
+        await queue_client.send_message(u'message5')
+        result = []
+        async for m in queue_client.receive_messages(max_messages=10):
+            result.append(m)
+
+        # Asserts
+        self.assertIsNotNone(result)
+        self.assertEqual(5, len(result))
+
+        for message in result:
+            self.assertIsNotNone(message)
+            self.assertNotEqual('', message.id)
+            self.assertNotEqual('', message.content)
+            self.assertNotEqual('', message.pop_receipt)
+            self.assertEqual(1, message.dequeue_count)
+            self.assertNotEqual('', message.inserted_on)
+            self.assertNotEqual('', message.expires_on)
+            self.assertNotEqual('', message.next_visible_on)
+
+    @QueuePreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_get_messages_with_page_bigger_than_max(self, storage_account_name, storage_account_key):
+        # Action
+        qsc = QueueServiceClient(self.account_url(storage_account_name, "queue"), storage_account_key)
+        queue_client = self._get_queue_reference(qsc)
+        await queue_client.create_queue()
+        await queue_client.send_message(u'message1')
+        await queue_client.send_message(u'message2')
+        await queue_client.send_message(u'message3')
+        await queue_client.send_message(u'message4')
+        await queue_client.send_message(u'message5')
+        result = []
+        async for m in queue_client.receive_messages(messages_per_page=5, max_messages=2):
```

Comment:
Maybe this case shouldn't be allowed and should raise a `ValueError`? I guess it "works" but also is a weird thing for the customer to do. I'll leave it up to you.

If you did check and raise an error, I would suggest doing that in the client method itself not the iterator.

Reply:
I agree that this functionality should not be expected. I think when I originally wrote this test and it passed, I made a judgement call and said "well, if the customer REALLY wants to do that, be my guest 😝" but yeah I think raising a `ValueError` here would be the right call!